File: my-folder/1468-check-if-n-and-its-double-exist/solution.c
bool checkIfExist(int* arr, int arrSize) {
    for (size_t i = 0; i < arrSize-1; i++)
        for (size_t j = i + 1; j < arrSize; j++){
            if ((arr[i] == 2 * arr[j]) || (2 * arr[i] == arr[j]))
                return true;
        }
    return false;
}

File: my-folder/1468-check-if-n-and-its-double-exist/solution.rs
impl Solution {
    pub fn check_if_exist(arr: Vec<i32>) -> bool {
        use std::collections::HashSet;
        let mut seen = HashSet::new();

        for &num in &arr {
            if seen.contains(&(2 * num)) || (num % 2 == 0 && seen.contains(&(num / 2))) {
                return true;
            }
            seen.insert(num);
        }

        false
    }
}
